modules/sdk-coin-avaxp/src/lib/utxoEngine.ts

```diff
@@ -0,0 +1,117 @@
+import { DecodedUtxoObj, SECP256K1_Transfer_Output } from './iface';
+import { BN, Buffer as BufferAvax } from 'avalanche';
+import { Signature } from 'avalanche/dist/common';
+import utils from './utils';
+import { BuildTransactionError } from '@bitgo/sdk-core';
+import { StandardAmountInput, StandardTransferableInput } from 'avalanche/dist/common/input';
+
+export interface InputData {
+  amount: BN;
+  txidBuf: BufferAvax;
+  outputidx: BufferAvax;
+  signareIdx: number[];
+  signatures: Signature[];
+}
+
+/**
+ * Inputs can be controlled but outputs get reordered in transactions
+ * In order to make sure that the mapping is always correct we create an addressIndx which matches to the appropiate
+ * signatureIdx
+ * @param {TransferableInput[]} utxos as transaction ins.
+ * @returns the list of UTXOs
+ */
+export function recoverUtxos(utxos: StandardTransferableInput[]): DecodedUtxoObj[] {
+  return utxos.map((utxo) => {
+    const secpInput = utxo.getInput() as StandardAmountInput;
+
+    // use the same addressesIndex as existing ones in the inputs
+    const addressesIndex: number[] = secpInput.getSigIdxs().map((s) => s.toBuffer().readUInt32BE(0));
+
+    return {
+      outputID: SECP256K1_Transfer_Output,
+      outputidx: utils.cb58Encode(utxo.getOutputIdx()),
+      txid: utils.cb58Encode(utxo.getTxID()),
+      amount: secpInput.getAmount().toString(),
+      threshold: addressesIndex.length,
+      addresses: [], // this is empty since the inputs from deserialized transaction don't contain addresses
+      addressesIndex,
+    };
+  });
+}
+
+/**
+ * Convert Utxos into inputs data. Input Objects changes regarding chains. This method return a plain object to be mapped late in chain input.
+ * Sender is a list of owners address and utxo address must contains all of them.
+ * Signers is a list of sender cat it in threshold size. Firsts senders are the signers.
```

Comment:
typo on `sender cat it`?

Reply:
https://github.com/BitGo/BitGoJS/blob/6d023175be2d689b599bbc8e62e71c713fa7934a/modules/sdk-coin-avaxp/src/lib/utxoEngine.ts#L45